sql/sql_class.cc

```diff
@@ -576,6 +579,22 @@ void THD::set_transaction(Transaction_ctx *transaction_ctx) {
   m_transaction.reset(transaction_ctx);
 }
 
+void THD::set_db_metadata() {
+  if (m_db.length) {
```

Comment:
If m_db.length is 0 (e.g. rpl_slave.cc calls to thd->reset_db(NULL_CSTR);), doesn't db_metadata.clear() also need to be called?